src/signing.rs

```diff
@@ -1,22 +1,22 @@
-use ring::hmac::{Algorithm, Key as HMACKey};
+use hmac::{Hmac, Mac};
+use sha2::Sha256;
 
-static HS256: Algorithm = ring::hmac::HMAC_SHA256;
-pub struct Signature(pub Vec<u8>);
+type HmacSha256 = Hmac<Sha256>;
+pub struct Signature(pub [u8; 32]);
 
 pub const AES_KEY_LEN: usize = 32;
 
-impl Signature {
-    pub fn as_bytes(&self) -> &[u8] {
-        &self.0
-    }
-}
 pub fn sign_hs256(key: [u8; AES_KEY_LEN], payload: &[u8]) -> Signature {
-    let hmac_key = HMACKey::new(HS256, &key);
-    Signature(ring::hmac::sign(&hmac_key, payload).as_ref().to_vec())
+    let mut mac = HmacSha256::new_from_slice(&key)
+        .expect("The key has a fixed size, which is known to be good.");
```

Comment:
This expect message is confusing to me. Are you saying this as a "can't happen"?

Reply:
Yes. If someone broke the code they'd get that message, but any size key is ok for Hmac.